faceted_search/src/index.js

```diff
@@ -25,41 +45,128 @@ const fields = [
+	// stateless, so state mutating operations need to be contained here.  In our case the sequence is
+	// (1) User clicks something in the UI (like enables or disables a search facet )
+	// (2) UI refreshes
+	// …some time passes
+	// (3) This gets called, and we write back the current query string to the browser's location using setSearchParams
+    useEffect(() => {
+		// For now, encode only the selected search facets and start page in the searchParams
+		let searchFields = encode(JSON.stringify(store.getState()['query']['searchFields']));
+		let start = encode(JSON.stringify(store.getState()['query']['start']/store.getState()['query']['rows']));
+		let sortFields = encode(JSON.stringify(store.getState()['query']['sortFields']));
+		let searchParamsDict = {"searchFields": searchFields, "start": start, "sortFields": sortFields};
+
+		// Update the query parameters with the latest values selected in the UI
+		setSearchParams(searchParamsDict);
+		
+    },[searchParams, store.getState()]);
+    
+    return (
+        <div>
+            <SolrFacetedSearch
+                        {...store.getState()}
+                        {...solrClient.getHandlers()}
+                        bootstrapCss={true}
+						customComponents={iSamples_componentPack}
+                        onSelectDoc={(doc) => {console.log(doc); }} />
+        </div>
+    );
+}
+
 // Register your app with the store
 store.subscribe(() =>
-	// In stead of using the handlers passed along in the onChange callback of SolrClient
-	// use the .getHandlers() method to get the default click / change handlers
-	ReactDOM.render(
-
-		<SolrFacetedSearch
-			{...store.getState()}
-			{...solrClient.getHandlers()}
-			bootstrapCss={true}
-			onSelectDoc={(doc) => console.log(doc)}
-		/>,
-		document.getElementById("app")
-	)
+	// The inclusion of the HashRouter and Routes wrapping our APP is what allows the searchParams functionality to work.
+    ReactDOM.render(
+        <HashRouter>
+            <Routes>
+                <Route path="/" element={<APP />}/>
+            </Routes>
+        </HashRouter>
+        ,
+        document.getElementById("app")
+    )
 );
 
+export const appendAnalytics = () => {
+    const script = document.createElement("script");
+    script.defer = true;
+    script.src = config.analytics_src;
+    script.setAttribute("data-domain", config.analytics_domain);
+    document.head.appendChild(script);
+}
+
 document.addEventListener("DOMContentLoaded", () => {
-	// this will send an initial search initializing the app
-	
-	solrClient.initialize();
+    // this will send an initial search initializing the app
+	// We just need to set state when we firstly open the page with url
+	// So, we only need to set the initalize solrClient rather than set them in the useEffect
+	// Get the parameters when the page loads. 
+	// let [searchParams, setSearchParams] = useSearchParams();
+	// console.log(searchParams)
+	let CurURL = window.location.href;
```

Comment:
In general, we shouldn't have variables starting with capital letters, suggest `curURL`.

Reply:
Generally, class names should be capitalized and variables should be camel-cased `curURL` or underscore separated (`cur_url`).  The official python standard asks for underscore separated, I think we're free to choose whatever we like for JavaScript.  The exception of course would be a symbolic constant, and those are typically all uppercase.

Reply:
This looks good re: best practices for React components: https://airbnb.io/javascript/react/

Reply:
We don't have to settle on something as part of this pull request, but we should settle on something and try to stick to it eventually.

Reply:
Looks like AirBNB has a regular JS style guide too: https://airbnb.io/javascript/